src/util/stats.js

```diff
@@ -1,48 +1,82 @@
-const SDC = require('statsd-client');
+const statsd = require('./statsd');
+const prometheus = require('./prometheus');
+const logger = require('../logger');
 
 const enableStats = process.env.ENABLE_STATS !== 'false';
-const statsServerHost = process.env.STATSD_SERVER_HOST || 'localhost';
-const statsServerPort = parseInt(process.env.STATSD_SERVER_PORT || '8125', 10);
-const instanceID = process.env.INSTANCE_ID || 'localhost';
-
-const statsdClient = new SDC({
-  host: statsServerHost,
-  port: statsServerPort,
-  prefix: 'transformer',
-  tags: {
-    instanceName: instanceID,
-  },
-});
+const statsClientType = process.env.STATS_CLIENT || 'statsd';
+
+let statsClient;
+function init() {
+  if (!enableStats) {
+    return;
+  }
+
+  if (statsClientType === 'statsd') {
+    statsClient = new statsd.Statsd();
+    logger.info('created statsd client');
+  } else if (statsClientType === 'prometheus') {
+    statsClient = new prometheus.Prometheus();
+    logger.info('created prometheus client');
+  } else {
+    logger.info("Invalid stats client type. Valid values are 'statsd' and 'prometheus'.");
+  }
+}
 
 // Sends the diff between current time and start as the stat
 const timing = (name, start, tags = {}) => {
-  if (enableStats) {
-    statsdClient.timing(name, start, tags);
+  if (!enableStats || !statsClient) {
+    return;
   }
-};
 
-const increment = (name, delta = 1, tags = {}) => {
-  if (enableStats) {
-    statsdClient.increment(name, delta, tags);
-  }
+  statsClient.timing(name, start, tags);
 };
 
-const decrement = (name, delta = -1, tags = {}) => {
-  if (enableStats) {
-    statsdClient.decrement(name, delta, tags);
+const increment = (name, tags = {}) => {
+  if (!enableStats || !statsClient) {
+    return;
   }
+
+  statsClient.increment(name, tags);
 };
 
 const counter = (name, delta, tags = {}) => {
-  if (enableStats) {
-    statsdClient.counter(name, delta, tags);
+  if (!enableStats || !statsClient) {
+    return;
   }
+
+  statsClient.counter(name, delta, tags);
 };
 
 const gauge = (name, value, tags = {}) => {
-  if (enableStats) {
-    statsdClient.gauge(name, value, tags);
+  if (!enableStats || !statsClient) {
```

Comment:
Let's have this evaluation at one place as this will be same for any kind of stat method. WDYT ?